src/main/java/TypewiseAlert/TypewiseAlert.java

```diff
@@ -16,72 +17,49 @@ public static BreachType inferBreach(double value, double lowerLimit, double upp
       }
       return BreachType.NORMAL;
     }
-    public enum CoolingType {
-      PASSIVE_COOLING,
-      HI_ACTIVE_COOLING,
-      MED_ACTIVE_COOLING
-    };
-    public static BreachType classifyTemperatureBreach(
-        CoolingType coolingType, double temperatureInC) {
-      int lowerLimit = 0;
-      int upperLimit = 0;
-      switch(coolingType) {
-        case PASSIVE_COOLING:
-          lowerLimit = 0;
-          upperLimit = 35;
-          break;
-        case HI_ACTIVE_COOLING:
-          lowerLimit = 0;
-          upperLimit = 45;
-          break;
-        case MED_ACTIVE_COOLING:
-          lowerLimit = 0;
-          upperLimit = 40;
-          break;
-      }
-      return inferBreach(temperatureInC, lowerLimit, upperLimit);
-    }
-    public enum AlertTarget{
-      TO_CONTROLLER,
-      TO_EMAIL
-    };
-    public class BatteryCharacter {
-      public CoolingType coolingType;
-      public String brand;
-    }
-    public static void checkAndAlert(
-        AlertTarget alertTarget, BatteryCharacter batteryChar, double temperatureInC) {
+    
+    
+    public static boolean checkAndAlert(
+        AlertTarget alertTarget, CoolingType coolingType, double temperatureInC) {
 
-      BreachType breachType = classifyTemperatureBreach(
-        batteryChar.coolingType, temperatureInC
-      );
+      BreachType breachType = inferBreach(temperatureInC, coolingType.getLowerLimit(), coolingType.getUpperLimit());
+      
+      return alertAction(breachType, alertTarget);
+    }
 
-      switch(alertTarget) {
+   
+	private static boolean alertAction(BreachType breachType, AlertTarget alertTarget) {
```

Comment:
The return type is boolean, but the function name is alertAction.
For a caller of this function, the meaning of true/false would be ambiguous. What would be the return if it failed to send (e.g., due to network failure?)
Always reconsider when you use primitive types to express 'meaningful' results